src/AWS.Logger.Core/Core/AWSLoggerCore.cs

```diff
@@ -122,36 +124,16 @@ public void AddMessage(string message)
             {
                 message = EMPTY_MESSAGE;
             }
-            if (_pendingMessageQueue.Count > _config.MaxQueuedMessages)
```

Comment:
I'm still going to insist we have some bounds check on this collection. One of the rules we have for this library is that it has as little impact on the running application as possible. In this code sample if I list a profile that is invalid the application eventually runs out of memory and dies. Same thing could happen if the IAM permission was removed to write to CloudWatch Logs.

```csharp
static void Main(string[] args)
{
    Console.WriteLine("Hello World!");
    try
    {
        var config = new AWS.Logger.AWSLoggerConfig
        {
            Profile = "fooo",
            Region = "us-east-1",
            LogGroup = "LoggingTestGroups"
                    
        };

        var factory = new LoggerFactory();

        factory.AddAWSProvider(config);

        var logger = factory.CreateLogger<Program>();

        while (true)
        {
            logger.LogError($"{DateTime.Now}: Another message");
        }
    }
    catch(Exception e)
    {
        Console.WriteLine(e.Message);
        Console.WriteLine(e.StackTrace);
    }
}
```

Reply:
Hi @normj could you do a correct test on NLog Multithreading part? I'm afraid not good to do this kind of test and result not correct.

Reply:
@KennethWKZ it isn't failing for me:
```
[9/5/2017 12:04:23 PM Informational] ------ Run test started ------
[9/5/2017 12:04:24 PM Informational] [xUnit.net 00:00:00.2473042]   Starting:    AWS.Logger.Log4Net.Tests
[9/5/2017 12:04:41 PM Informational] [xUnit.net 00:00:17.5216963]   Finished:    AWS.Logger.Log4Net.Tests
[9/5/2017 12:04:41 PM Informational] ========== Run test finished: 3 run (0:00:18.6229207) ==========
```

Reply:
@peterdeme Ya, same as me, result is fine. But i trying to find the maximum limit such as "out of memory" issue

Reply:
@normj That's weird, I thought it should throw exception. I'll check what's happening.